src/test/java/net/minecraftforge/debug/world/RaidEventTest.java

```diff
@@ -0,0 +1,96 @@
+package net.minecraftforge.debug.world;
+
+import net.minecraft.Util;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.entity.raid.Raid;
+import net.minecraft.world.entity.raid.Raider;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.world.RaidEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.Set;
+
+@Mod("raid_event_test")
+public class RaidEventTest
+{
+    public RaidEventTest()
+    {
+        MinecraftForge.EVENT_BUS.addListener(this::getCustomRaid);
+        MinecraftForge.EVENT_BUS.addListener(this::loadCustomRaid);
+        MinecraftForge.EVENT_BUS.addListener(this::onRaidSpawnGroup);
+        MinecraftForge.EVENT_BUS.addListener(this::onRaidStop);
+    }
+
+    private static final Random RAND = new Random();
+    public void getCustomRaid(RaidEvent.GetOrCreate event)
+    {
+        if (RAND.nextBoolean())
+            event.setCustomRaid((id) -> new TestModRaid(id, event.getLevel(), event.getPos()));
```

Comment:
For a test mod it is not a really good idea to make the effect change based. You coud be unlucky and not see anything for quite a few raids.